src/expr.cpp

```diff
@@ -9354,16 +9312,16 @@ bool FunctionSymbolExpr::ResolveOverloads(SourcePos argPos, const std::vector<co
         candidateExpandCosts.push_back(cost);
     }
 
-    // Find the best cost, and then the candidate or candidates that have
-    // that cost.
+    // Find the best cost, and then the candidate or candidates that have that cost.
     for (int i = 0; i < (int)candidateCosts.size(); ++i) {
         if (candidateCosts[i] != -1 && candidateCosts[i] < bestMatchCost) {
             bestMatchCost = candidateCosts[i];
         }
     }
+
     // None of the candidates matched
     if (bestMatchCost == (1 << 30)) {
-        return lReportErrorUnableToFindOverload(matches, argTypes, argCouldBeNULL, name, pos);
+        return bestMatchCost;
```

Comment:
As I understand the sanitizer output, the memory leaks happens here for memory allocated on line 9310 `int *cost = new int[argTypes.size()];`

Reply:
Fixed